job_creator.php

```diff
@@ -42,6 +42,23 @@ public function getInstallerVersion(): string
                 return $cmsMajor . '.' . $portions[1] . '.x-dev';
             }
         }
+        // has a lockstepped .x-dev requirement in composer.json
```

Comment:
This feels like it should be a fallback, after "hardcoded installer version for repo version" and probably after "fallback to use the next-minor or latest-minor version of installer" - because if we've hardcoded an installer version, we should use that.

Reply:
I've moved the block below "hardcoded installer version for repo version", but keeping it above "fallback to use the next-minor or latest-minor version of installer" because that fallback is really "i have no idea though I want to use something". When you're looking for lockstepeed versions of things in composer.json you have a very good idea of what to install at that point.

Reply:
I don't quite agree about what that fallback is there for - but since they're both exactly that (fallbacks) I have no issues with this order of operations.